progetto_esa.tex

```diff
@@ -101,33 +101,77 @@ \section{Compiling for qemu}
+In /board folder we can find the instructions to compile and run buildroot for all the devices supported.
+In particular, we have to consider the readme.md file that indicates the specific def\_config file to invoke to customize the compilation process.
 
 \section{Customizing and adding packages to buildroot}
+Buildroot is highly customizable and it allows to integrate in your distribution every kind of packages and kernel modules.
 you can place project-specific customizations in two locations:
-
-directly within the Buildroot tree, typically maintaining them using branches in a version control system so that upgrading to a newer Buildroot release is easy.
-outside of the Buildroot tree, using the br2-external mechanism. 
-This mechanism allows to keep package recipes, board support and configuration files outside of the Buildroot tree, while still having them nicely integrated in the build logic. 
+\begin{itemize}
+\item Directly within the Buildroot tree, typically maintaining them using branches in a version control system so that upgrading to a newer Buildroot release is easy.
+\item Outside of the Buildroot tree, using the br2-external mechanism. 
+This mechanism allows to keep package recipes, board support and configuration files outside of the Buildroot tree, while still having them nicely integrated in the build logic.
+\end{itemize} 
 We call this location a br2-external tree. 
 As described in the official manual, a br2-external tree must contain at least those three files:
-
 \begin{itemize}
-\item external.desc
-\item external.mk
-\item Config.in
+\item external.desc: which provides a name and a description;
+\item Config.in: configuration options that will be included in menuconfig;
+\item external.mk, will be included in the make logic;
 \end{itemize}
+
 If we would like to integrate packages,
 we have to consider two scenarios, some packages only contain a kernel module, other packages contain programs and libraries in addition to kernel modules. 
 Buildroot's helper infrastructure supports both cases.
+We have first to consider that a package is defined by a folder in package directory.
+
+
 In our case we intended to inlcude as an example to describe the customization, a package developed by Angelo Compagnucci, friend ans school collegue.
 The purpose of this package is to provide a software implementation of pwm.
+Le'ts now describe how to integrate a package.
+We can start saying that each package must contain these three files:
+\begin{itemize}
+\item Config.in;
+\item package makefile;
+\item package hash;
+\end{itemize}
+The file config.in, contains the necessary information for the menuconfig tool to make it available with all the detailed information.
+Here is the example for our package PWM\_GPIO. 
+\begin{lstlisting}
+config BR2_PACKAGE_PWM_GPIO
+	bool "pwm-gpio"
+	help
+	  Kernel driver for software pwm on gpio
+
+	  https://github.com/angeloc/pwm-gpio
+\end{lstlisting}
```

Comment:
Here you can say why you define a BR2_* symbol saying for example that it will be expanded by the kconfig system to a variable in .config (BR2_PACKAGE_PWM_GPIO=y). This variable will then be read by the makefile system to trigger the evaluation of the corresponding .mk file. This lets you explain how and why the Config.in is linked to the .mk file.